x-pack/plugins/security_solution/common/detection_engine/schemas/request/perform_bulk_action_schema.ts

```diff
@@ -102,11 +110,23 @@ const bulkActionEditPayloadRuleActions = t.type({
 
 export type BulkActionEditPayloadRuleActions = t.TypeOf<typeof bulkActionEditPayloadRuleActions>;
 
+const bulkActionEditPayloadSchedule = t.type({
+  type: t.literal(BulkActionEditType.set_schedule),
+  value: t.type({
+    interval,
+    meta: t.type({
+      from: t.string,
+    }),
+  }),
+});
```

Comment:
I don't think it's necessary to expose this `meta` nested rule structure here. This bulk action is concerned with intervals and lookback times, so I'd reflect it in the request body. Let's also make it more robust with the non-empty string requirement:

```ts
const bulkActionEditPayloadSchedule = t.type({
  type: t.literal(BulkActionEditType.set_schedule),
  value: t.type({
    interval: NonEmptyString,
    lookback: NonEmptyString,
  }),
});
```

Ideally, it would be great to validate that this string represents a valid duration (`1h`, `2s`, etc). I don't know if we have such validation elsewhere.

Reply:
I created [a new type `TimeLength` ](https://github.com/elastic/kibana/pull/140166/files#diff-7b6de91ddeb343c8d3bc13e5ef38365e1fce6056ed27045993160bcebccefc80) which validates this exact use case. However, I can't get VSCode to see the exported type out of the `@kbn/securitysolution-io-ts-types` package even though [I exported it from its export file.](https://github.com/elastic/kibana/pull/140166/files#diff-84fdb9989bcc937bd7c4431156b0502a50bfcd2866a848e2d1c1c3df579e8ce1) and re-bootstrapped the app.

Have you ever added a new io-type to this package? Any idea if I might be missing something?

Reply:
There is [a similar validator in alerting framework](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/common/parse_duration.ts), that is used for the same case, [validating interval value for bulk edit of schedules operation
](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/routes/bulk_edit_rules.ts#L40)

It would be ideal if we can use it, as this is validator on alerting side of interval values we pass to it

Reply:
Discussed with @banderror ; we'd rather have the validation in the io-ts type check as it is done as early as possible of the execution flow. Updated the type, now called TimeDuration, and corresponding tests.